packages/core/src/part.js

```diff
@@ -415,5 +415,8 @@ Part.prototype.setCutOnFold = function (p1, p2) {
   return this
 }
 
+Part.prototype.isEmpty = function() {
+  return Object.keys(this.paths).length === 0 && Object.keys(this.snippets).length === 0
+}
```

Comment:
yes, `part.render` is not set consistently throughout patterns. Most patterns return the part empty rather than explicitly hiding it if it's not called for due to settings. An example is the buttonPlacket part in Simon.

Reply:
I can check for more conditions, or we can find a different way to make sure that parts are hidden if they're empty, but I think that second option is not actually going to result in something different, it might just put it in a different place

Reply:
Should we then perhaps address that underlying issue?
Related: We could perhaps add support for `conditionalParts` who, much like conditional plugins, use a run-time check to see whether they should bother adding themselves to the pattern at all.

Perhaps we could also add a test for this to our auto-generated pattern unit tests?

Reply:
I'd love to address the underlying issue, and I think a unit test is a great way to make sure it's happening. I forgot about those and figured that there wasn't really a way to set that as a standard

maybe instead of `conditionalParts`, there's a `shouldInclude` function in the part config?